pkg/cmd/cli/cmd/env.go

```diff
@@ -232,6 +233,24 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 	if !one && len(resourceVersion) > 0 {
 		return cmdutil.UsageError(cmd, "--resource-version may only be used with a single resource")
 	}
+	// Keep a copy of the original objects prior to updating their environment.
+	// Used in constructing the patch(es) that will be applied in the server.
+	oldObjects, err := resource.AsVersionedObjects(infos, clientConfig.Version)
+	if err != nil {
+		return err
+	}
+	if len(oldObjects) != len(infos) {
+		return fmt.Errorf("expected to change %d object to versioned, changed %d", len(infos), len(oldObjects))
```

Comment:
`"could not convert all objects to API version %s", clientConfig.Version`?
not sure what circumstances could lead here, so I'm not sure what I'd recommend the user do, but at least call out the problematic version?